flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -260,6 +287,7 @@ public void onContainersAllocated(List<Container> containers) {
 			numPendingContainerRequests = Math.max(0, numPendingContainerRequests - 1);
 			log.info("Received new container: {} - Remaining pending container requests: {}",
 					container.getId(), numPendingContainerRequests);
+			workerNodeMap.put(container.getId().toString(), new YarnWorkerNode(container));
```

Comment:
Here we should create a `ResourceID` to use it as the key. This `ResourceID` must then be send to the launched TaskManager such that it uses it upon registration.